### DISCUSSION TITLE
Allow to publish server endpoints and key via DNS

### DISCUSSION THREAD

**Discussion by bt90** [rustdesk/rustdesk#1268] [General]
Currently a user has to input the hbbs address and key in the settings page or receive a specially named rustdesk.exe. Both of these options are rather cumbersome.

Ideally the user would just need to enter the domain of the rustdesk server e.g `rustdesk.example.com` and all necessary information is retrieved via DNS SRV(hbbs+hbbr addresses) and TXT(key) records.

**Comment by rustdesk** [comment]
#1423

**Comment by xrgzs** [comment]
🤔 I am eager to access my self-hosted rustdesk hbbr server via SRV records, as its dynamically changing 🔢 port number poses a challenge. But, alas! There's no documentation 📚 referencing or guiding through SRV records access.

After diving deep into the repo 🔍, I came up empty-handed – no code relating to SRV record parsing in sight. I stumbled upon a discussion, hinted at from issue #1423. It’s disheartening 😔 to realize almost two years have drifted by since that issue shifted to a discussion, seemingly forgotten. This situation leaves much to be desired.😥

My sincere wish is for the acceleration 🚀 of developing this vital feature – **embracing SRV records for both hbbs and hbbr server addresses**. And, let's not forget, a documentation refresh 🔄 to align with these advancements once they're live.

Here's to hoping for a brighter, more connected future! ✨

@rustdesk
